src/sfdxProject.ts

```diff
@@ -78,6 +124,42 @@ export class SfdxProjectJson extends ConfigFile<ConfigFile.Options> {
     Object.assign(defaultOptions, options || {});
     return defaultOptions;
   }
+
+  /**
+   * Validates sfdx-project.json against the schema.
+   *
+   * ***See*** [sfdx-project.schema.json] (https://raw.githubusercontent.com/forcedotcom/schemas/master/schemas/sfdx-project.schema.json)
+   */
+  public async schemaValidate(): Promise<void> {
+    if (!this.hasRead) {
+      // read calls back into this method after setting this.hadRead = true
+      await this.read();
```

Comment:
This is a bit dangerous since `read()` conditionally calls `schemaValidate()` but `this.hasRead` should be true in that case.  It would only cause a problem if `read` were changed to catch any errors from the super and continue.  However, maybe throwing an error instead of reading here is safer.